### PR TITLE
Reset top banner local storage entry every 30 days

### PR DESCRIPTION
**Pull request by dgarcia360**
Closes https://github.com/scylladb/sphinx-scylladb-theme/issues/639

- Splits the JS file in different modules for better maintenance.
- Sets an expiry date of 30 days for the top banner.

## How to test this PR

1. Clone this PR.
2. In `docs/conf.py`, set  `html_theme_options.hide_banner` to `"false"`.
3. On terminal window, go to the `docs` folder.
4. Run `make preview`
5. Open http://localhost:5500/. The site should render without errors.
6. Close the top banner.
7. Open the Inspector. In Chrome, right click > **Inspect**.
8. Go to **Application** > **Local Storage** tab.
9. Select **localhost:5050**.
10. You should see the entry "scylla-docs-hide-banner" with an expiry date.

image

When a page is rendered, the theme checks[ if the current date is greater than the expiry date](https://github.com/scylladb/sphinx-scylladb-theme/pull/651/files#diff-38ac84ba1b1c772b4d6fdfeefaeb8ad0c8a3d3caf329593b0b179c7155c8cbadR21-R24) to invalidate the entry.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -98,7 +98,7 @@
 html_theme_options = {
     "conf_py_path": "docs/source/",
     "hide_edit_this_page_button": "false",
-    "hide_banner": "true",
+    "hide_banner": "false",
```
Comment (tzach) [inline docs/source/conf.py:101]: why updating the deafult value?

File: sphinx_scylladb_theme/static/js/main.bundle.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
...
...=j.find.attr(t,e))?void 0:i)},attrHooks:{type:{set:function(t,e){if(!m.radioValue&&"radio"===e&&z(t,"input")){var n=t.value;return t.setAttribute("type",e),n&&(t.value=n),e}}}},removeAttr:function(t,e){var n,i=0,o=e&&e.match(q);if(o&&1===t.nodeType)for(;n=o[i++];)t.removeAttribute(n)}}),he={set:function(t,e,n){return!1===e?j.removeAttr(t,n):t.setAttribute(n,n),n}},j.each(j.expr.match.bool.source.match(/\w+/g),(function(t,e){var n=pe[e]||j.find.attr;pe[e]=function(t,e,i){var o,r,s=e.toLowerCase();return i||(r=pe[s],pe[s]=o,o=null!=n(t,e,i)?s:null,pe[s]=r),o}}));var ve=/^(?:input|select|textarea|button)$/i,me=/^(?:a|area)$/i;function ge(t){return(t.match(q)||[]).join(" ")}function ye(t){return t.getAttribute&&t.getAttribute("class")||""}function be(t){return Array.isArray(t)?t:"string"==typeof t&&t.match(q)||[]}j.fn.extend({prop:function(t,e){return Q(this,j.prop,t,e,arguments.length>1)},removeProp:function(t){return this.each((function(){delete this[j.propFix[t]||t]}))}}),j.extend({prop:function(t,e,n){var i,o,r=t.nodeType;if(3!==r&&8!==r&&2!==r)return 1===r&&j.isXMLDoc(t)||(e=j.propFix[e]||e,o=j.propHooks[e]),void 0!==n?o&&"set"in o&&void 0!==(i=o.set(t,n,e))?i:t[e]=n:o&&"get"in o&
```
Comment (tzach) [inline sphinx_scylladb_theme/static/js/main.bundle.js:2]: no idea what this change is

File: src/js/promo-banner.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    $(".secondary-side-nav").css("margin-top", promoBannerHeight);
+    $(".layout").addClass("layout--has-banner");
+  } else {
+    promoBanner.hide();
+  }
+};
+
+export const onCloseBanner = () => {
+  $(".promo-banner__close").on("click", function () {
+    setItemWithExpiry(localStorageKey, "1", 30);
```
Comment (tzach) [inline src/js/promo-banner.js:44]: can we move this value to config.py?

File: docs/source/conf.py (hunk 1 above)
Comment (dgarcia360) [reply docs/source/conf.py]: For demo purposes. Added back the previous value.

File: sphinx_scylladb_theme/static/js/main.bundle.js (hunk 2 above)
Comment (dgarcia360) [reply sphinx_scylladb_theme/static/js/main.bundle.js:2]: This is the minified js. For more information, see https://sphinx-theme.scylladb.com/stable/contribute/contribute-theme.html#building-the-frontend

**Review by annastuchlik — APPROVED** [review]
LGTM.
I've followed the testing procedure in the description of this PR.

**Comment by annastuchlik** [comment]
This PR can be merged.

File: src/js/promo-banner.js (hunk 3 above)
Comment (dgarcia360) [reply src/js/promo-banner.js:44]: It is possible. If necessary, we can work on it as part of a separate issue.

**Pull request merged**